tfhe/docs/fundamentals/serialization.md

```diff
@@ -108,54 +65,150 @@ use tfhe::{
+    safe_serialize(&ct, &mut buffer, 1 << 20).unwrap();
+
+    assert!(safe_deserialize_conformant::<FheUint8>(buffer.as_slice(), 1 << 20, &conformance_params_2)
         .is_err());
 
-    let ct2 = DeserializationConfig::new(1 << 20)
-        .deserialize_from::<FheUint8>(buffer.as_slice(), &conformance_params_1)
+    let ct2: FheUint8 = safe_deserialize_conformant(buffer.as_slice(), 1 << 20, &conformance_params_1)
         .unwrap();
 
     let dec: u8 = ct2.decrypt(&client_key);
     assert_eq!(msg, dec);
-    
-    
+
+
     // Example with a compact list:
     let msgs = [27, 188u8];
     let mut builder = CompactCiphertextList::builder(&public_key);
     builder.extend(msgs.iter().copied());
     let compact_list = builder.build();
 
     let mut buffer = vec![];
-    SerializationConfig::new(1 << 20).serialize_into(&compact_list, &mut buffer).unwrap();
-    
+    safe_serialize(&compact_list, &mut buffer, 1 << 20).unwrap();
+
     let conformance_params = CompactCiphertextListConformanceParams {
         shortint_params: params_1.to_shortint_conformance_param(),
         num_elements_constraint: ListSizeConstraint::exact_size(2),
     };
-    DeserializationConfig::new(1 << 20)
-        .deserialize_from::<CompactCiphertextList>(buffer.as_slice(), &conformance_params)
+    safe_deserialize_conformant::<CompactCiphertextList>(buffer.as_slice(), 1 << 20, &conformance_params)
         .unwrap();
 }
 ```
+
+The safe serialization and deserialization use `bincode` internally.
+
+To selectively disable some of the features of the safe serialization, you can use `SerializationConfig`/`DeserializationConfig` builders.
+For example, it is possible to disable the data versioning:
+
+```rust
+// main.rs
+
+use tfhe::safe_serialization::{safe_deserialize_conformant, SerializationConfig};
+use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+use tfhe::ServerKey;
+use tfhe::{generate_keys, ConfigBuilder};
+
+fn main() {
+    let params_1 = PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+
+    let config = ConfigBuilder::with_custom_parameters(params_1).build();
+
+    let (client_key, server_key) = generate_keys(config);
+
+    let mut buffer = vec![];
+
+    SerializationConfig::new(1 << 30).disable_versioning().serialize_into(&server_key, &mut buffer).unwrap();
+
+    // You will still be able to load this item with `safe_deserialize_conformant`, but only using the current version of TFHE-rs
+    let _server_key_deser: ServerKey =
+        safe_deserialize_conformant(buffer.as_slice(), 1 << 30, &config.into()).unwrap();
+}
+```
+
+## Serialization/deserialization using serde
+
+**TFHE-rs** uses the [Serde](https://crates.io/crates/serde) framework and implements Serde's `Serialize` and `Deserialize` traits.
+
+This allows you to serialize into any [data format](https://serde.rs/#data-formats) supported by serde. In the following example, we use [bincode](https://crates.io/crates/bincode) for its binary format.
+
+Here is a full example:
+
```

Comment:
maybe explain that TFHE-rs does nothing for the user to guarantee type compatibility here